### PR TITLE
Fix lint errors, improve linting, enable travis.

### PR DESCRIPTION
**Pull request by vext01**
DON'T MERGE. TESTING.

### CODE REVIEW

**Comment by vext01** [comment]
Try this Laurie. The last commit will be reverted and is only included to show that travis flags a lint error.

Fixing the lint errors was very tedious, so I welcome travis wholeheartedly!

File: softdev.bib [hunk 1]
```diff
@@ -15,7 +15,7 @@
 @inproceedings{Khadka14how,
     author={Khadka, Ravi and Batlajery, Belfrit V. and Saeidi, Amir M. and Jansen, Slinger and Hage, Jurriaan},
     title={How Do Professionals Perceive Legacy Systems and Software Modernization?},
-    booktitle={Proceedings of the 36th International Conference on Software Engineering},
+    booktitle={International Conference on Software Engineering},
     series={{ICSE} 2014},
```
Comment (ltratt) [inline softdev.bib]: I think this one should just be `booktitle={ICSE}` (and no `series`)?

File: softdev.bib [hunk 2]
```diff
@@ -441,15 +443,15 @@ @techreport{kalibera12quantifying
...
+    booktitle = "Multi-Language Object Models",
     year = "1994",
     month = Aug
 }
 
 @inproceedings{kaplan98idls,
     author = {Alan Kaplan and John Ridgway and Jack C. Wileden},
     title = {Why {IDLs} Are Not Ideal},
-    booktitle = {9th International Workshop on Software Specification and Design},
+    booktitle = {Proceedings of the Conference on Software Specification and Design},
```
Comment (ltratt) [inline softdev.bib]: Should this just be "Conference on Software Specification and Design"?

File: softdev.bib [hunk 3]
```diff
@@ -1394,17 +1418,23 @@ @misc{barrett16warmup
     howpublished     = {http://arxiv.org/abs/1602.00602}
 }
 
+- year is from the git import date of the doc file
```
Comment (vext01) [inline softdev.bib]: Not this bib comment. Correct?

File: softdev.bib [hunk 4]
```diff
@@ -1394,17 +1418,23 @@ @misc{barrett16warmup
...
 @misc{pstate,
     title = {Intel {P-state driver}, {Linux} Kernel Documentation},
     howpublished = {\href{https://www.kernel.org/doc/Documentation/cpu-freq/intel-pstate.txt}{https://www.kernel.org/doc/Documentation/cpu-freq/intel-pstate.txt}},
-    note = {Accessed: 2016-01-15}
+    note = {Accessed: 2016-01-15},
+    year = {2014},
+    author = {The Linux Kernel Developers},
 }
 
+- year is from the git import date of the doc file
```
Comment (vext01) [inline softdev.bib]: And here.

File: softdev.bib [hunk 5]
```diff
@@ -1467,11 +1499,13 @@ @inproceedings{bolz16sqpyte
     url       = {http://arxiv.org/abs/1503.08623}
 }
 
+ - year is from the first date covered by the CLBG license
```
Comment (vext01) [inline softdev.bib]: And here.

**Comment by ltratt** [comment]
This looks good, modulo a couple of small comments and one over-arching one. I'm fairly sure that months should be of the format `month = Jan` i.e. without quotes or curlies. I think these are macros that bibtex formats then expand as appropriate. But if you do `month = {jan}` I think it always gets printed as "jan" (lower-case!) which isn't what we want. Warning: I might be wrong.

File: softdev.bib (hunk 3 above)
Comment (ltratt) [inline softdev.bib]: Yes, I think it's correct.

File: softdev.bib (hunk 2 above)
Comment (vext01) [inline softdev.bib]: That's a deliberate lint error I introduced!

Comment (ltratt) [inline softdev.bib]: Ah... well at least you know I did a proper review ;)

**Comment by vext01** [comment]
I think you are right:
http://nokyotsu.com/latex/bibtex.html#month

**Comment by vext01** [comment]
Seems like prebib is emitting all months in braces. Probably not what we want.

Sadly the bibtex parser doesn't differentiate `{Jan}` from `Jan` either, so I might have to manually scan the file with regex looking for bad quoting. Boo.

Will look more into this after food.

**Comment by ltratt** [comment]
OK.

**Comment by vext01** [comment]
Ready for re-review. Fixed all comments, fixed months, checked months, travis now runs pdflatex to check it builds. OK?

File: bin/prebib [hunk 6]
```diff
@@ -104,6 +138,11 @@ class PreBib:
         filtered_data = { k : v for (k, v) in entry.items() if k not in excludes }
         self.regex_warn(key, filtered_data)
         self.missing_warn(bclass, key, filtered_data)
+
+        if "month" in entry:
```
Comment (ltratt) [inline bin/prebib]: Probably simpler as `if "month" in entry and entry["month"] not in OK_MONTHS`.

Comment (vext01) [inline bin/prebib]: fixed.

**Comment by ltratt** [comment]
OK. I think it fair to say that some squashing is probably needed ;) Then I can merge.

**Comment by vext01** [comment]
Here we go.

**Pull request merged**